jwst/pipeline/calwebb_spec2.py

```diff
@@ -142,7 +142,12 @@ def process_exposure_product(
         # a grism image, if so get the catalog
         # name from the asn and record it to the meta
         if exp_type in WFSS_TYPES:
-            input.meta.source_catalog.filename = members_by_type['sourcecat'][0]
+            if input.meta.source_catalog.filename is None:
```

Comment:
This logic gives precedence to the keyword value. I think we should have the value in the ASN file take precedence, since that was the original design. So look for that first and then if it doesn't exist, try to get it from the keyword. As soon as a user runs the pipeline once on a given exposure, the keyword will then be populated and hence upon subsequent executions (if they rerun it) it was always default to using the old keyword value. I think it should come from the ASN, when available.

Reply:
okay, I will reverse the logic an make the asn take precedence.